addons/hw_drivers/controllers/driver.py

```diff
@@ -1,6 +1,9 @@
 #!/usr/bin/python3
 import logging
 import time
+import inspect
+from abc import ABCMeta
```

Comment:
why this change?

Reply:
At first there was some code in the base scale supported method, which raised an exception when it was tested. Adding the abstractmethod decorator to supported and checking for abstract classes to keep base classes out of the drivers list was a way to avoid that. It is not needed anymore but I figured it made sense to keep base classes out of the list, so supported methods can be partially implemented or call not implemented methods. What do you think?

Reply:
@pha-odoo You should just check if 'supported' is implemented to detect if you are in real driver, as implementing it is the only way to have deveice that works.